src/main/java/com/iemr/hwc/service/user/UserServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.iemr.hwc.service.user;
+
+import com.iemr.hwc.data.location.*;
+import com.iemr.hwc.data.login.Users;
+import com.iemr.hwc.data.login.UsersMasterVillage;
+import com.iemr.hwc.repo.location.*;
+import com.iemr.hwc.repo.login.UserLoginRepo;
+import com.iemr.hwc.repo.login.UserMasterVillageRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.sql.Timestamp;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+    @Autowired
+    private UserLoginRepo userLoginRepo;
+
+    @Autowired
+    private UserMasterVillageRepo userMasterVillageRepo;
+
+    @Autowired
+    private DistrictBranchMasterRepo districtBranchMasterRepo;
+
+    public String setMasterVillage(Long userID, Integer villageID){
+        String response = "";
+        UsersMasterVillage usersMasterVillage = userMasterVillageRepo.getByUserIDAndVillageID(userID, villageID);
+        if(usersMasterVillage==null || (usersMasterVillage!=null && usersMasterVillage.getActive()==false)){
+            DistrictBranchMapping districtBranchMapping = districtBranchMasterRepo.findByDistrictBranchID(villageID);
+            Users userBD = userLoginRepo.getUserByUserID(userID);
+            if(userBD!=null) {
+                if(districtBranchMapping!=null){
+                    UsersMasterVillage user = new UsersMasterVillage();
+                    user.setUser(userBD);
+                    user.setMasterVillage(districtBranchMapping);
+                    user.setActive(true);
+                    user.setCreatedDate(new Timestamp(System.currentTimeMillis()));
+                    user.setLastModDate(new Timestamp(System.currentTimeMillis()));
+                    user = userMasterVillageRepo.save(user);
+                    if(user!=null){
+                        response = "ok";
+                    }
+                    else{
+                        response = "ko";
```

Comment:
what is this ko? change it to reasonable string

Reply:
Change done. It's change to "not_ok" @roopesh-beehyv